src/safeposix/syscalls/fs_calls.rs

```diff
@@ -799,7 +799,10 @@ impl Cage {
 
                     let mut nonblocking = false;
                     if pipe_filedesc_obj.flags & O_NONBLOCK != 0 { nonblocking = true;}
-                    return pipe_filedesc_obj.pipe.write_to_pipe(buf, count, nonblocking) as i32
+                    
+                    let retval = pipe_filedesc_obj.pipe.write_to_pipe(buf, count, nonblocking) as i32;
+                    if retval == -(Errno::EPIPE as i32) { interface::kill_inner(self.cageid, 13); } // Trigger SIGPIPE
+                    retval
```

Comment:
I believe retval has to be outside the end bracket here, does this compile?

Reply:
This looks fine to me actually, the retval here is just an implicit return, and putting it outside the end bracket would make the match statement break.

Reply:
Yeah the other branches of the match statement also use implicit returns. So this should work.